app/upgrades.go

```diff
@@ -175,6 +177,23 @@ func (app *App) registerUpgrade0_3_0Rc8(_ upgradetypes.Plan) {
 	)
 }
 
+func (app *App) registerUpgrade0_4_0(upgradeInfo upgradetypes.Plan) {
+	const planName = "v0.4.0"
+	app.UpgradeKeeper.SetUpgradeHandler(planName,
+		func(ctx context.Context, plan upgradetypes.Plan, fromVM module.VersionMap) (module.VersionMap, error) {
+			sdkCtx := sdk.UnwrapSDKContext(ctx)
+			sdkCtx.Logger().Info("Upgrade handler execution", "name", planName)
+			return app.mm.RunMigrations(ctx, app.configurator, fromVM)
+		},
+	)
+
+	app.storeUpgrade(planName, upgradeInfo, storetypes.StoreUpgrades{
```

Comment:
Can you add a reference to where you found the directions for the circuit module upgrade?